internal/services/servicebus/servicebus_topic_resource.go

```diff
@@ -283,6 +283,27 @@ func resourceServiceBusTopicCreateUpdate(d *pluginsdk.ResourceData, meta interfa
 		return fmt.Errorf("retrieving ServiceBus Namespace %q (Resource Group %q): %+v", id.NamespaceName, id.ResourceGroupName, err)
 	}
 
+	isPremiumNamespacePartitioned := true
+	var sku namespaces.SkuName
+	if nsModel := resp.Model; nsModel != nil {
+		sku = nsModel.Sku.Name
+	}
+
+	if sbNamespaceModel := resp.Model; sbNamespaceModel != nil {
+		if sbNamespaceModel.Properties != nil &&
+			sbNamespaceModel.Properties.PremiumMessagingPartitions != nil && *sbNamespaceModel.Properties.PremiumMessagingPartitions == 1 {
+			isPremiumNamespacePartitioned = false
+		}
+	}
+
+	if sku == namespaces.SkuNamePremium {
+		if isPremiumNamespacePartitioned && !enablePartitioning {
+			return fmt.Errorf("non-partitioned entities are not allowed in partitioned namespace")
+		} else if !isPremiumNamespacePartitioned && enablePartitioning {
+			return fmt.Errorf("the parent premium namespace is not partitioned and the partitioning for premium namespace is only available at the namepsace creation")
+		}
```

Comment:
I don't understand what the desired outcome is here. If the parent namespace is not partitioned but the topic is partitioned does the API reject this?

Reply:
Yes, the api behavior changed. It didn't throw below error before:
![image](https://github.com/user-attachments/assets/a41a691f-11d8-4e80-9834-0e4e10c8ac83)

Do we still need to add the validation in the provider, given the fact that the api throws the error now. 

The situation for servicebus topic is the same as the servicebus queue, for servicebus queue, we have the validation as :
![image](https://github.com/user-attachments/assets/13936e91-53ba-4fae-ad10-fcda5292d889)


I was trying to do the same thing for the topic in this pr.

Reply:
The API error in your screenshots is being handled on lines
https://github.com/hashicorp/terraform-provider-azurerm/blob/f1fd24c1a60279952f072aaaa50c79811d8ff243/internal/services/servicebus/servicebus_topic_resource.go#L300-L301

This review comment refers to the check and error message on lines
https://github.com/hashicorp/terraform-provider-azurerm/blob/f1fd24c1a60279952f072aaaa50c79811d8ff243/internal/services/servicebus/servicebus_topic_resource.go#L302-L303
Which now checks the inverse, i.e. if the parent namespace is not partitioned but the topic is partitioned. So to repeat and rephrase my question above:

Does the API reject the request if the *topic is partitioned* but the *parent namespace is not partitioned*?

Reply:
For servicebus namespace, the namespace topic/ queue is created under servicebus namespace. For premium namespace, as mentioned in the [doc ](https://learn.microsoft.com/en-us/azure/service-bus-messaging/enable-partitions-premium): "We do not allow creating non-partitioned entities in a partitioned namespace."
![image](https://github.com/user-attachments/assets/98576a59-2359-4145-b966-b1d908639e10)
But for servicebus queue/ topic, if user tries to create the non-partitioned queue/ topic under a partitioned namespace by setting the ` enable_partitioning = true`, the api will set the value to `false` without throwing any error. The next time user performs the `terraform plan`, there will be diff as :
![image](https://github.com/user-attachments/assets/430fbec5-4e1b-4471-b1c0-7344acd3b1a8)

hope it makes sense to you..